src/tests/fuzz/fuzz_applayerparserparse.c

```diff
@@ -0,0 +1,101 @@
+/**
+ * @file
+ * @author Philippe Antoine <contact@catenacyber.fr>
+ * fuzz harness for AppLayerProtoDetectGetProto
+ */
+
+
+#include "suricata-common.h"
+#include "app-layer-detect-proto.h"
+#include "flow-util.h"
+#include "app-layer-parser.h"
+
+#define HEADER_LEN 6
+
+void fuzz_openFile(const char * name) {
+}
+
+AppLayerParserThreadCtx *alp_tctx = NULL;
+
+int LLVMFuzzerTestOneInput(const uint8_t *data, size_t size) {
+    Flow * f;
+    TcpSession ssn;
+
+    if (size < HEADER_LEN) {
+        return 0;
+    }
+
+    if (alp_tctx == NULL) {
```

Comment:
you could use an init function i.e. int LLVMFuzzerInitialize(int *argc, char ***argv)

Reply:
Yes but that is not recommended by libfuzzer : http://llvm.org/docs/LibFuzzer.html#startup-initialization

Reply:
Ok, you are right. I prefer this kind of solution